tests/trace/dsl/test_query.py

```diff
@@ -1,302 +1,768 @@
+            "name": ["root span", "embedding span", "retriever span", "llm span"],
+        }
+    ).set_index("context.span_id")
+    actual = sq(session, project_name="abc")
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
 
-    for field in (SPAN_ID, f"{CONTEXT_PREFIX}{SPAN_ID}"):
-        project = Projection(field)
-        assert project(spans[0]) == "0"
-        assert project(spans[1]) == "1"
+    sq = SpanQuery().select("span_id").with_index("trace_id")
+    expected = pd.DataFrame(
+        {
+            "context.trace_id": ["012", "012", "012", "012"],
+            "context.span_id": ["234", "345", "456", "567"],
+        }
+    ).set_index("context.trace_id")
+    actual = sq(session, project_name="abc")
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1).sort_values("context.span_id"),
+        expected.sort_index().sort_index(axis=1).sort_values("context.span_id"),
+    )
+    del sq, actual, expected
 
-    for field in (INPUT_VALUE, f"{ATTRIBUTE_PREFIX}{INPUT_VALUE}"):
-        project = Projection(field)
-        assert project(spans[0]) == "000"
-        assert project(spans[1]) is None
 
-    for field in (RETRIEVAL_DOCUMENTS, f"{ATTRIBUTE_PREFIX}{RETRIEVAL_DOCUMENTS}"):
-        project = Projection(field)
-        assert project(spans[0]) == []
-        assert project(spans[1]) == [
-            {
-                DOCUMENT_CONTENT: "10",
-                DOCUMENT_SCORE: 100,
-            }
-        ]
+def test_select_nonexistent(session: Session) -> None:
+    sq = SpanQuery().select("name", "opq", "opq.rst")
+    expected = pd.DataFrame(
+        {
+            "context.span_id": ["234", "345", "456", "567"],
+            "name": ["root span", "embedding span", "retriever span", "llm span"],
+            "opq": [None, None, None, None],
+            "opq.rst": [None, None, None, None],
+        }
+    ).set_index("context.span_id")
+    actual = sq(session, project_name="abc")
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
 
 
-def test_concatenation(spans):
-    concat = Concatenation("12")
-    assert list(concat(spans[2])) == [("12", "1\n\n2")]
+def test_default_project(session: Session) -> None:
+    sq = SpanQuery().select(
+        "name",
+        **{"Latency (milliseconds)": "latency_ms"},
+    )
+    expected = pd.DataFrame(
+        {
+            "context.span_id": ["2345"],
+            "name": ["root span"],
+            "Latency (milliseconds)": [30000.0],
+        }
+    ).set_index("context.span_id")
+    actual = sq(session, root_spans_only=True)
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
```

Comment:
It's just to clean up for the next test so it's not inadvertently re-using anything. Obviously it's not needed if there's only one test, but I left it there just in case (it's also just copied-pasted anyway).

Reply:
If it's needed, I think it's a sign that the test case should be split up. It's confusing to include it at the end of a test case when it's not needed, I think. If some cleanup process is needed, we should try to run it as a fixture when possible.